### PR TITLE
Remove messaging providers/interfaces

### PR DESCRIPTION
**Pull request by HaoK**
Mostly just a lot of deleting code...

Fixes https://github.com/aspnet/Identity/issues/359

cc @divega @suhasj

### CODE REVIEW

File: samples/IdentitySample.Mvc/Startup.cs [hunk 1]
```diff
@@ -86,39 +80,5 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
             SampleData.InitializeIdentityDatabaseAsync(app.ApplicationServices).Wait();
         }
 
-        public class EmailMessageProvider : IIdentityMessageProvider
```
Comment (suhasj) [inline samples/IdentitySample.Mvc/Startup.cs]: nit: should these be is separate classes ?

Comment (HaoK) [inline samples/IdentitySample.Mvc/Startup.cs]: I'm not sure what you are asking here, this code is deleted

Comment (suhasj) [inline samples/IdentitySample.Mvc/Startup.cs]: ah my bad... one of those brain drain days

**Comment by HaoK** [comment]
Merged c2e96fa570593688ccd159372562f6bbe545b640

**Pull request closed without merging**